api/v1/storagecluster_types.go

```diff
@@ -377,6 +377,31 @@ type MultiCloudGatewaySpec struct {
 	// +nullable
 	// +optional
 	DisableLoadBalancerService bool `json:"disableLoadBalancerService,omitempty"`
+
+	// Allows Noobaa to connect to an external Postgres server
+	// +optional
+	ExternalPgConfig ExternalPGSpec `json:"externalPgConfig,omitempty"`
+}
+
+type ExternalPGSpec struct {
+	// Enables Noobaa to use external Postgres
+	// +optional
+	Enable bool `json:"enable,omitempty"`
+	// PGSecret stores the secret name which contains connection string of the Postgres server
+	// +optional
+	PGSecret string `json:"pgSecret,omitempty"`
+	// PGSecretNamespace is the namespace where secret lives
+	// +optional
+	PGSecretNamespace string `json:"pgSecretNamespace,omitempty"`
+	// AllowSelfSignedCerts will allow the Postgres server to use self signed certificates to authenticate
+	// +optional
+	AllowSelfSignedCerts bool `json:"allowSelfSignedCerts,omitempty"`
+	// TLSSecret stores the secret name which contains the client side certificates if enabled
+	// +optional
+	TLSSecret string `json:"tlsSecret,omitempty"`
+	// TLSecretNamespace is the namespace where the TLS secret lives
+	// +optional
+	TLSSecretNamespace string `json:"tlsSecretNamespace,omitempty"`
```

Comment:
The namespace mentioned in the spec is the same namespace where storagecluster CR is created or it can be anything?

Reply:
It can be anything. Noobaa takes in a secret reference so name and namespace can be anything.

Reply:
i dont know if we need to support this one or not, i assume we always want the users to make use of the namespace where storagecluster is created /odf is installed.

Reply:
let's remove the namspace I will do the same for NooBaa repo as well. We can't mount secrets from different namespaces anyway.

Reply:
Also PGSecretNamespace should be removed

Reply:
@vbnrh will you be deleting these values from the types?

Reply:
@Madhu-1 @jackyalbo I had failed to see these reviews. I will update the values